Assets/ControllerNotesInput.cs

```diff
@@ -0,0 +1,62 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using System;
+using UnityEngine.UI;
+
+public class ControllerNotesInput : MonoBehaviour {
+
+    public InputField lastWordInput;
+    public Button[] noteButtons;
+    public Button doneInput;
+
+    private Stack<int> registeredIndices = new Stack<int>(); //Keep a record of registered indices, so that we can push the cancel button to delete the text
+    private bool isInputDone = false;
+    private string[] letters = new string[] { "A", "N", "N", "I", "E" };
+
+
+    void Update() {
+        Debug.Log("Update is running");
+        //If B is pressed then we should cancel a letter
+        if (Input.GetButtonDown("ghostMode")) {
+            Debug.Log("Canceling");
+            if (registeredIndices.Count != 0) {
+                cancelLetter(registeredIndices.Pop());
+            }
+        }
+    }
+
+    //Getter used to check if the player is done entering input with the notes.
+    // Used from the story scene manager. 
+    public bool getIsInputDone() { return isInputDone; }
+
+    //Used when the Done? button is clicked
+    public void setInputDone () {
+        isInputDone = true;
+    }
+
+    public void enterLetter(int index) {
+        //concat the letter to the input text
+        lastWordInput.text += letters[index];
+        //disable the input button so that the user cannot use the same input twice 
+        noteButtons[index].interactable = false;
+        //We should select the next note as the selected buttonk, before disabling the current button
+        bool foundSelectableNote = false;
```

Comment:
So this is for when your selecting the notes for input, when you click the note, we disable the note so that you can't reclick the same note. When we do this we need to find the next note to select. So we start from the current note index and search through to the end to see if there is any other note to select, if we find a note we stop searching and select this note. If we don't find a note, we select the done button.